core/extension/gdextension_interface.h

```diff
@@ -2268,6 +2280,34 @@ typedef GDExtensionObjectPtr (*GDExtensionInterfaceObjectGetInstanceFromId)(GDOb
  */
 typedef GDObjectInstanceID (*GDExtensionInterfaceObjectGetInstanceId)(GDExtensionConstObjectPtr p_object);
 
+/**
+ * @name object_has_script_method
+ * @since 4.3
+ *
+ * Checks if this object has a script with the given method.
+ *
+ * @param p_object A pointer to the Object.
+ * @param p_method A pointer to a StringName identifying the method.
+ *
+ * @returns true if the method exists.
+ */
+typedef GDExtensionBool (*GDExtensionInterfaceObjectHasScriptMethod)(GDExtensionConstObjectPtr p_object, GDExtensionConstStringNamePtr p_method);
```

Comment:
> Checks if this object has a script with the given method.

My understanding from this documentation is:
- if no script is attached, this always returns `false`
- if a user defines _any_ method in the script with that name -- whether it overrides a virtual method or not -- it returns true
- unclear whether it applies to `_ready`, `_process` and so on?

Is this correct? Maybe we can clarify these semantics a bit?
Also if/how it relates to [`ClassDB::has_method()`](https://docs.godotengine.org/en/stable/classes/class_classdb.html#class-classdb-method-class-has-method) 🤔

Reply:
That is correct! Regarding:

> * unclear whether it applies to `_ready`, `_process` and so on?

It applies to those as well. If the script has any method by the given name, it returns `true`.

> Also if/how it relates to [`ClassDB::has_method()`](https://docs.godotengine.org/en/stable/classes/class_classdb.html#class-classdb-method-class-has-method) 🤔

`ClassDB::has_method()` is looking for method binds, aka native methods, not script methods.

Anyway, I'll see if I can update the docs to clarify this a bit!

Reply:
Looks good! About `_ready` etc, I assume it only returns true if they are actually _overridden_ in the script, not just if the type offers such a method -- correct?

That's the only unclarity I could see remaining, but if you think it's clear enough as-is, that's also fine with me.

Reply:
> About `_ready` etc, I assume it only returns true if they are actually _overridden_ in the script, not just if the type offers such a method -- correct?

Yes, it only returns true if that specific script has an implementation for the named method. It doesn't care if there is a virtual method registered for that type (which I think we don't even track in release builds anyway)

> That's the only unclarity I could see remaining, but if you think it's clear enough as-is, that's also fine with me.

Personally, I think it's clear enough, so I'll mark as resolved. But we can always add more to the docs if we end up getting questions about this :-)